src/main/java/com/shapesecurity/salvation/directiveValues/HostSource.java

```diff
@@ -128,4 +129,58 @@ public boolean matchesOnlyOrigin(@Nonnull SchemeHostPortTriple origin) {
             (isDefaultPort ? "" : ":" + (this.port == Constants.WILDCARD_PORT ? "*" : this.port)) +
             (this.path == null ? "" : this.path);
     }
+
+    public static boolean pathMatches(@Nullable String pathA, @Nullable String pathB) {
+
+        if (pathA == null || pathA.isEmpty()) {
+            return true;
+        }
+
+        if (pathB == null || pathB.isEmpty()) {
+            return false;
+        }
+
+        if (pathA.matches("/") && pathB.isEmpty()) {
+            return true;
+        }
+
+        boolean exactMatch = pathA.endsWith("/") ? false : true;
+
+        List<String> pathListA = Arrays.asList(pathA.split("/"));
+        List<String> pathListB = Arrays.asList(pathB.split("/"));
+
+        long countA = pathA.chars().filter(ch -> ch =='/').count();
```

Comment:
@michaelficarra , welcome to improve the count logic. Didn't have much time for more elegant solution, but the idea is that strict splitting loses info on delimiters, and just counting them should be enough